### PR TITLE
rework package to be importable (proof of concept)

### LINKED ISSUE #3: improvements: make this library importable instead of forcing it to be the top level.
still playing with the code, but think this should work closer to how go-gl/glfw works where you can just import and go.

- figure out a way of statically linking the flutter engine.
- remove main function and expose a top level package function that perform the same work.

been working on getting these changes to work running into issues with the shared library, which i haven't had a chance to track down. basically go is having trouble loading the library but isn't being clear on where its looking.

### PR DESCRIPTION
**Pull request by james-lawrence**
- don't mind the top level package name =) just didn't want to type out `go-flutter-desktop-embedder` 
- moves example into its own folder/package.
- adds a .build directory that ignores the .so library so cleanup isn't really required.
- removes the requirement for the user to add the flutter engine .so inside the package.
  but requires that the user of the library provide the load path for the flutter engine.
- exposes some options and a top level Run function.

after this changes I was able to get my flutter project to load and display. wasn't entirely functional but was a decent POC.

fixes #3

### CODE REVIEW

**Comment by Drakirus** [comment]
GJ, I like It.
Your example works great.

From what I see, to be able to build or run we have to specify the `CGO_LDFLAGS` which I am not a big fan. Could it be possible to add at the top of the 'example/stocks/main.go' the cgo ldflags ?  
*something like*
```go
// #cgo linux LDFLAGS: -L${SRCDIR}/flutter/library/linux
import "C"
```

About the name since this implementation is bound to glfw. Could we name the library `flutter-glfw` ?

**Comment by james-lawrence** [comment]
not a huge fan of the CGO_LDFLAGS either, I thought about the cgo option as well, though I havent had a chance to test it. might work, still find either solution less than satisfactory as it put requirements on the working directory.

either way, both cases are up to the end user to choose to use, so if it your way works no problem putting it in the example.

as for the name: we should avoid -'s in the name past experience has should those to be problematic/non-obvious when attempting to actually use the package.

File: flutter/build.go [hunk 1]
```diff
@@ -4,7 +4,12 @@ package flutter
...
 #cgo linux CFLAGS: -I${SRCDIR}/library
-#cgo linux LDFLAGS: -L${SRCDIR}/library/linux -lflutter_engine -Wl,-rpath,$ORIGIN/flutter/library/linux
+#cgo linux LDFLAGS: -lflutter_engine -Wl,-rpath,flutter/library/linux
 
 */
 import "C"
+
+import (
+	// prevents dep from stripping out the c source files in flutter library.
+	_ "github.com/Drakirus/go-flutter-desktop-embedder/flutter/library"
```
Comment (Drakirus) [inline flutter/build.go]: i just learned something new.
Very useful!

**Review by pchampio — APPROVED** [review]

**Comment by Drakirus** [comment]
I will merge it soon.

File: flutter/build.go (hunk 1 above)
Comment (james-lawrence) [reply flutter/build.go]: useful in this case being a nasty hack because dep does the wrong thing lol

**Pull request merged**